### PR TITLE
Add a reftest that entering fullscreen mode takes over the window

### LINKED ISSUE #22804: Add reftests for WebVR/XR
Once we have a test WebVR/XR display (#22795) we can write reftests which use it.

### PR DESCRIPTION
**Pull request by asajeffrey**
Add a baseline reftest for fullscreen that ensures that going fullscreen actually uses the whole window. This uses a pref that requests that fullscreen mode takes over the window rather than the whole screen. The pref doesn't do anything right now, but is a bit of future-proofing for when we do have proper fullscreen.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #22804.
- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22809)

### CODE REVIEW

File: tests/wpt/web-platform-tests/fullscreen/reftests/fullscreen-baseline.html
Comment (jdm) [inline tests/wpt/web-platform-tests/fullscreen/reftests/fullscreen-baseline.html]: This won't work in non-servo browsers which require user activation to enter fullscreen.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jugglinmike** [comment]
[We'd like to get testdriver.js working in reftests for cases like these](https://github.com/web-platform-tests/wpt/issues/13183), but we're not there yet.

@jdm the tests in this patch have [already been submitted to WPT](https://github.com/web-platform-tests/wpt/pull/15218) and [labeled as "already reviewed downstream"](https://github.com/web-platform-tests/wpt/pull/15218#pullrequestreview-199307957). Is that a bug?

**Comment by jdm** [comment]
No, it's how our system works. It gets merged upstream after it merges in this repository.

**Comment by jugglinmike** [comment]
Ah, got it. Maybe we should change @wpt-pr-bot's message a bit and/or delay approval.

File: tests/wpt/web-platform-tests/fullscreen/reftests/fullscreen-baseline.html
Comment (asajeffrey) [reply tests/wpt/web-platform-tests/fullscreen/reftests/fullscreen-baseline.html]: True, but we'd need other browsers to add support for yesterday which request fullscreen but expect to reuse the current window. Such a testing pref could also allow fullscreen without user engagement.

**Comment by jdm** [comment]
I'll point out that the Chromium upstreaming system behaves the exact same way. The presence if the 'do not merge' label acts as an indicator of the status.

**Comment by asajeffrey** [comment]
Should I add a README about what's needed to pass this test?

**Comment by jdm** [comment]
No. We shouldn't add upstream tests that other browsers can't pass.

**Comment by asajeffrey** [comment]
Okay, I'll move the reftest into the wpt/mozilla directory then. At some point the other browsers might like some reftests for fullscreen, but we can deal with that when there's an agreed way to do the testing.

**Review by jdm — APPROVED** [review]

**Comment by asajeffrey** [comment]
Argh I meant r=jdm, oops.

**Pull request merged**
